### ISSUE TITLE
Incorrect Reading of Punctuation Marks in Text-to-Speech Synthesis

### ISSUE THREAD

**Issue by MmdrezaMolavi** [rhasspy/piper#122]
Hello! I have noticed an issue with the text-to-speech (TTS) synthesis in which long sentences, such as paragraphs, are not being properly read out. It appears that the generated audio does not acknowledge punctuation marks, leading to an unnatural and disjointed reading experience.

Steps to reproduce:

1.Provide a long sentence or a paragraph of text for TTS synthesis.
2.Listen to the generated audio output.

Expected behavior:
The TTS system should accurately read the text and pause appropriately at punctuation marks, such as commas, periods, and question marks. This allows for a more coherent and natural listening experience.

Actual behavior:
When synthesizing long sentences or paragraphs, the TTS system fails to recognize and properly read the punctuation marks. As a result, the generated audio lacks the necessary pauses and intonation, making it difficult to comprehend and follow the text.

Additional information:
I tested the en-us-ryan-high and en-us-lessac-medium speakers.

**Comment by MmdrezaMolavi** [comment]
@beqabeqa473 In fact, when I run it in the bash mode, the pronunciation and pauses are correct when encountering punctuation marks. However, the issue arises when I use Python run command.

**Comment by beqabeqa473** [comment]
correct, python_run uses espeak-ng installed systemwide, which doesn't
implement changes from michael.

On 7/3/23, MmdrezaMolavi ***@***.***> wrote:
> @beqabeqa473 In fact, when I run it in the bash mode, the pronunciation and
> pauses are correct when encountering punctuation marks. However, the issue
> arises when I use Python run command.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/rhasspy/piper/issues/122#issuecomment-1618273500
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by set-soft** [comment]
> correct, python_run uses espeak-ng installed systemwide, which doesn't implement changes from michael.

But even installing the patched espeak-ng lib won't work because the Python code doesn't implement the correct phonemizer. The Python code always use `espeak_phonemizer.Phonemizer` and it will drop the punctuation, even for the patched lib.

I think the solution is what I implemented in #138 

I just took the code @synesthesiam wrote to check if the modified lib works (is Python code) and put it in the python_run code (well, is a little bit more than just this). I was able to get the punctuation in the Python output.

**Comment by set-soft** [comment]
> This is also very dependent on the voice itself. Some voice datasets do not include punctuation, so the model can't learn when to do short vs. long pauses.

Do you know which are the more expressive models for English?
I want to set them as defaults in the plug-in for Ooba Gooba Text Generator WebUI that I'm creating.
Here is an example of what I'm doing:

https://github.com/rhasspy/piper/assets/22823627/34857cec-d377-489b-9ee3-5cef34466acf

This example is using _en_US-amy-medium_ for the girl and *en_US-libritts-high* (p922) for the description.

**Comment by goldyfruit** [comment]
Having to install a custom `espeak-ng` version is a bummer but I least now I got the answer why punctuation is not followed. :+1: 

Thanks

**Comment by rmcpantoja** [comment]
In the recent updates, you need piper_phonemize to fix this in python_run.

**Comment by goldyfruit** [comment]
> In the recent updates, you need piper_phonemize to fix this in python_run.

So when using `piper_phonemize` I don´t  need eSpeak forks?

**Comment by goldyfruit** [comment]
> > In the recent updates, you need piper_phonemize to fix this in python_run.
> 
> So when using `piper_phonemize` espeak-ng fork is not required?

Just tested, installing `piper_phonemize` only doesn't fix this punctuation issue.

**Comment by rmcpantoja** [comment]
> > In the recent updates, you need piper_phonemize to fix this in python_run.
> 
> So when using `piper_phonemize` espeak-ng fork is not required?

No, isn't required. Piper_phonemize includes it.



> > > In the recent updates, you need piper_phonemize to fix this in python_run.
> > 
> > 
> > So when using `piper_phonemize` espeak-ng fork is not required?
> 
> Just tested, installing `piper_phonemize` only doesn't fix this punctuation issue.

Make sure you have the latest version of the piper API (python_run) which adds support for piper_phonemize.
